modules/kameleonvolume/kameleonvolumereader.cpp

```diff
@@ -37,48 +37,97 @@ namespace {
 }
 
 namespace openspace {
+namespace kameleonvolume {
 
 KameleonVolumeReader::KameleonVolumeReader(const std::string& path)
     : _path(path)
 {
     if (!FileSys.fileExists(path)) {
-        LERROR(_path << "does not exist");
-        return;
+        LERROR(_path << " does not exist");
+        throw ghoul::FileNotFoundError(_path);
     }
 
     long status = _kameleon.open(_path);
     if (status != ccmc::FileReader::OK) {
-        LERROR("Failed to open file " << _path << " with kameleon");
+        LERROR("Failed to open file " << _path << " with Kameleon");
+        throw ghoul::RuntimeError("Failed to open file: " + _path + " with Kameleon");
         return;
     }
 
     _model = _kameleon.model;
+
+    // Possibly use a kameleon interpolator instead of a model interpolator?
     _interpolator = std::unique_ptr<ccmc::Interpolator>(_model->createNewInterpolator());
 }
 
-std::unique_ptr<RawVolume<float>> KameleonVolumeReader::readFloatVolume(
+std::unique_ptr<volume::RawVolume<float>> KameleonVolumeReader::readFloatVolume(
+    const glm::uvec3 & dimensions,
+    const std::string & variable,
+    const glm::vec3 & lowerDomainBound,
+    const glm::vec3 & upperDomainBound) const
+{
+    float min, max;
+    return readFloatVolume(dimensions, variable, lowerDomainBound, upperDomainBound, min, max);
+}
+
+std::unique_ptr<volume::RawVolume<float>> KameleonVolumeReader::readFloatVolume(
     const glm::uvec3 & dimensions,
     const std::string & variable,
     const glm::vec3 & lowerBound,
-    const glm::vec3 & upperBound) const
+    const glm::vec3 & upperBound,
+    float& minValue,
+    float& maxValue) const
 {
-    auto volume = std::make_unique<RawVolume<float>>(dimensions);
+    minValue = FLT_MAX;
+    maxValue = FLT_MIN;
+
+    auto volume = std::make_unique<volume::RawVolume<float>>(dimensions);
     const glm::vec3 dims = volume->dimensions();
     const glm::vec3 diff = upperBound - lowerBound;
 
-    _model->loadVariable(variable);
+    std::function<float(const std::string&, glm::vec3)> interpolate =
```

Comment:
Better to specify this as `auto`, which will probably be more efficient since it will be stored as an internal struct with `operator()` overload rather than a `std::function` object